js/components/authoring.tsx

```diff
@@ -106,14 +125,54 @@ export default class Authoring extends PureComponent<IProps, IState> {
     this.toggleAdvancedOptions = this.toggleAdvancedOptions.bind(this);
   }
 
+  updateColorMapOptions(_config: Record<string, any>) {
+    const availableColorMaps = getAvailableColorMaps(_config);
+    // default colormap options should be limited to allowed options
+    DROPDOWN_OPTIONS.colormap = availableColorMaps;
```

Comment:
There were two reasons I didn't pursue this path originally:
1. I was trying to avoid major refactoring if possible
2. There's an implicit assumption throughout this component that there's a one-to-one correspondence between state variables and `config` variables which would be broken if we start introducing additional component state variables.

Yes, I'm sure we could avoid the `forceUpdate()` with some additional refactoring, putting things in state, perhaps separating the `config` state from the component state somehow. Alternatively, we could rename these module variables so they're not ALL_CAPS and figure that one `forceUpdate()` isn't that big a deal in the context of an authoring page that as you say, only a few people will ever see. The question is whether it's worth it.